packages-content-model/roosterjs-content-model-api/lib/modelApi/selection/adjustTrailingSpaceSelection.ts

```diff
@@ -0,0 +1,65 @@
+import { createText } from 'roosterjs-content-model-dom';
+import { iterateSelections } from 'roosterjs-content-model-core';
+import type {
+    ContentModelDocument,
+    ContentModelSegment,
+    ContentModelText,
+} from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export function adjustTrailingSpaceSelection(model: ContentModelDocument) {
+    iterateSelections(model, (_, __, block, segments) => {
+        if (block?.blockType == 'Paragraph') {
+            const tempSegments = [...block.segments];
+            tempSegments?.forEach((segment, index) => {
+                if (
+                    segment.isSelected &&
+                    segment.segmentType == 'Text' &&
+                    hasTrailingSpace(segment.text) &&
+                    !isTrailingSpace(segment.text)
+                ) {
+                    splitTextSegment(block.segments, segment, index);
+                }
+            });
+        }
+        return true;
```

Comment:
In a multiple paragraph selection, if we return true here, we would only fix the first paragraph. Should we instead return false to continue iterating the next paragraphs?

Reply:
good point!!